packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -909,6 +996,50 @@ void ArbCore::operator()() {
     core_machine->abortMachine();
 }
 
+bool ArbCore::runMachineWithMessages(MachineExecutionConfig& execConfig,
+                                     size_t max_message_batch_size,
+                                     std::unique_ptr<MachineThread>& machine) {
+    ReadSnapshotTransaction tx(data_storage);
+    auto messages_result = readNextMessages(
+        tx, machine->machine_state.output.fully_processed_inbox,
+        max_message_batch_size);
+    if (!messages_result.status.ok()) {
+        core_error_string = messages_result.status.ToString();
+        machine_error = true;
+        std::cerr << "ArbCore failed getting message entry: "
+                  << core_error_string << "\n";
+        return false;
+    }
+
+    if (!messages_result.data.empty()) {
+        execConfig.inbox_messages = messages_result.data;
+
+        auto success = machine->runMachine(execConfig);
+        if (!success) {
+            core_error_string = "Error starting machine thread";
+            machine_error = true;
+            std::cerr << "ArbCore error: " << core_error_string << "\n";
+            return false;
+        }
+
+        if (delete_checkpoints_before_message != uint256_t(0)) {
+            /*
+            deleteOldCheckpoints(delete_checkpoints_before_message,
+                                 save_checkpoint_message_interval,
+                                 ignore_checkpoints_after_message);
+            */
+            ignore_checkpoints_after_message = 0;
+            save_checkpoint_message_interval = 0;
+            delete_checkpoints_before_message = 0;
+        }
+    } else {
+        // Machine all caught up, no messages to process
+        machine_idle = true;
```

Comment:
Setting a member variable here seems dangerous to me, as this might not be operating on the core machine, and this member variable specifically refers to the core machine.

Reply:
Updated so that core_machine is always used.